.github/workflows/main.yml

```
@@ -98,6 +98,19 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
+
+    - name: Cache datasets
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-datasets
+      with:
+        path: |
+          data
+          ~/tensorflow_datasets
+        key: build-${{ env.cache-name }}-${{ github.ref }}
+        restore-keys: |
+          build-${{ env.cache-name }}-
```

I think we can even simplify as it should not change between the PR, the datasets are pretty constants.
```suggestion
        key: build-${{ env.cache-name }}
        restore-keys: |
          build-${{ env.cache-name }}
```